CRM/MembershipExtras/Upgrader.php

```diff
@@ -309,27 +309,14 @@ private function copyLastInstalmentLineItemsToRecurContrib($paymentPlan, $lineIt
     }
   }
 
-  private function getCustomFieldId($customGroupName, $customFieldName) {
-    return civicrm_api3('CustomField', 'getvalue', [
-      'return' => 'id',
-      'custom_group_id' => $customGroupName,
-      'name' => $customFieldName,
-    ]);
-  }
-
   private function createCustomValueForPaymentPlan($paymentPlanId) {
-    $nextPeriodCustomFieldId = $this->getCustomFieldId('related_payment_plan_periods', 'next_period');
-    $prevPeriodCustomFieldId = $this->getCustomFieldId('related_payment_plan_periods', 'previous_period');
+    $custom_fields = array (
+      'previous_period' => 0,
+      'next_period' => 0
+    );
+    $params = array('entityID' => $paymentPlanId, $custom_fields);
 
-    civicrm_api3('ContributionRecur', 'create', [
-      [
-        'id' => $paymentPlanId,
-        'custom_' . $nextPeriodCustomFieldId => 0,
-      ], [
-        'id' => $paymentPlanId,
-        'custom_' . $prevPeriodCustomFieldId => 0,
-      ],
-    ]);
+    CRM_Core_BAO_CustomValueTable::setValues($params);
```

Comment:
Hmm, the way it was before kind of could have worked too, except on back-end, but the params array that was being sent is kind of weird. So this should work if you do this too:

```php
civicrm_api3('ContributionRecur', 'create', [
  'id' => $paymentPlanId,
  'custom_' . $nextPeriodCustomFieldId => 0,
  'custom_' . $prevPeriodCustomFieldId => 0,
]);
```

Reply:
This doesn't work as it should. It attempts to create a contribution with empty contact_id and frequency_interval which was causing the fatal error.

Reply:
Yeah, because there were 2 arrays being sent, which is something back-end can't deal with. But if yu just send the one array, with 'id' and 'custom_xxxx' fields, it should work ok.

Wrong:
```php
// This effectively would try to create a recurring contribution, as CiviCRM isn't able to find the 'id' field. There is also no contact_id, causing the error. This array is actually similar to sending an empty array into the API call, as CiviCRM back-end can't deal with that kind of input.
[
      [
        'id' => $paymentPlanId,
        'custom_' . $nextPeriodCustomFieldId => 0,
      ], [
        'id' => $paymentPlanId,
        'custom_' . $prevPeriodCustomFieldId => 0,
      ],
    ]
```

Right:
```php
// ID is given, sent as a key within the array, uses that ID to update the existing recurring contribution.
[
  'id' => $paymentPlanId,
  'custom_' . $nextPeriodCustomFieldId => 0,
  'custom_' . $prevPeriodCustomFieldId => 0,
]
```